presto-hive/src/test/java/com/facebook/presto/hive/HiveTestUtils.java

```diff
@@ -267,4 +269,28 @@ public static Slice longDecimal(String value)
     {
         return encodeScaledValue(new BigDecimal(value));
     }
+
+    public static Optional<String> getProperty(String name)
+    {
+        String systemPropertyValue = System.getProperty(name);
+        String environmentVariableValue = System.getenv(name);
+        if (systemPropertyValue == null) {
+            if (environmentVariableValue == null) {
+                return Optional.empty();
+            }
+            else {
+                return Optional.of(environmentVariableValue);
+            }
+        }
+        else {
```

Comment:
IMO the nesting of conditionals is confusing to read. Here's how I would structure it:

```java
Optional<String> systemPropertyValue = Optional.ofNullable(System.getProperty(name));
Optional<String> environmentVariableValue = Optional.ofNullable(System.getenv(name));

if (!systemPropertyValue.isPresent() && !environmentVariableValue.isPresent()) {
    return Optional.empty();
}

if (!systemPropertyValue.isPresent() && environmentVariableValue.isPresent()) {
    return environmentVariableValue;
}

if (!systemPropertyValue.equals(environmentVariableValue)) {
  throw new ....
}

return systemPropertyValue;
```

Reply:
@ZacBlanco In this way the systemPropertyValue will be checked multiple times while my original code does not do the if check more than necessary. It would also create the Optional objects more than necessary. In the past I've seen big perf and reliability problems by creating Optional objects too many and too fast. But since this is just a one-time off thing I'm good with both ways. I'll update the PR using your rewrite.

Reply:
@ZacBlanco @aditi-pandit This block is missing one check and it should be like this:
```
public static Optional<String> getProperty(String name)
    {
        Optional<String> systemPropertyValue = Optional.ofNullable(System.getProperty(name));
        Optional<String> environmentVariableValue = Optional.ofNullable(System.getenv(name));

        if (!systemPropertyValue.isPresent() && !environmentVariableValue.isPresent()) {
            return Optional.empty();
        }

        if (!systemPropertyValue.isPresent() && environmentVariableValue.isPresent()) {
            return environmentVariableValue;
        }

        if (environmentVariableValue.isPresent() && !systemPropertyValue.equals(environmentVariableValue)) {.  // need to have environmentVariableValue.isPresent() && 
            throw new IllegalArgumentException(format("%s is set in both Java system property and environment variable, but their values are different. The Java system property value is %s, while the" +
                            " environment variable value is %s. Please use only one value.",
                    name,
                    systemPropertyValue,
                    environmentVariableValue));
        }

        return systemPropertyValue;
    }
```
I think this is actually harder to understand than my original code. What do you think?